boa_engine/src/syntax/ast/function/async_arrow_function.rs

```diff
@@ -0,0 +1,124 @@
+use super::FormalParameterList;
+use crate::syntax::ast::{
+    expression::{Expression, Identifier},
+    join_nodes, ContainsSymbol, StatementList,
+};
+use boa_interner::{Interner, ToIndentedString};
+
+/// An async arrow function expression, as defined by the [spec].
+///
+/// An [async arrow function][mdn] expression is a syntactically compact alternative to a regular function
+/// expression. Arrow function expressions are ill suited as methods, and they cannot be used as
+/// constructors. Arrow functions cannot be used as constructors and will throw an error when
+/// used with new.
+///
+/// [spec]: https://tc39.es/ecma262/#prod-AsyncArrowFunction
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
+#[cfg_attr(feature = "deser", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, PartialEq)]
+pub struct AsyncArrowFunction {
+    name: Option<Identifier>,
+    parameters: FormalParameterList,
+    body: StatementList,
+}
+
+impl AsyncArrowFunction {
+    /// Creates a new `AsyncArrowFunction` AST Expression.
+    #[inline]
+    pub(in crate::syntax) fn new(
+        name: Option<Identifier>,
+        params: FormalParameterList,
```

Comment:
Would it make sense to rename this to "parameters" to remove the name repetition below?